src/vs/workbench/services/editor/browser/editorOverrideService.ts

```diff
@@ -40,18 +43,37 @@ export class EditorOverrideService extends Disposable implements IEditorOverride
 	readonly _serviceBrand: undefined;
 
 	private _contributionPoints: Map<string | glob.IRelativePattern, ContributionPoints> = new Map<string | glob.IRelativePattern, ContributionPoints>();
+	private static readonly overrideCacheStorageID = 'editorOverrideService.cache';
+	private cache: Set<string> | undefined;
 
 	constructor(
 		@IEditorGroupsService private readonly editorGroupService: IEditorGroupsService,
 		@IConfigurationService private readonly configurationService: IConfigurationService,
 		@IQuickInputService private readonly quickInputService: IQuickInputService,
 		@INotificationService private readonly notificationService: INotificationService,
 		@ITelemetryService private readonly telemetryService: ITelemetryService,
+		@ILifecycleService private readonly lifeCycleService: ILifecycleService,
+		@IStorageService private readonly storageService: IStorageService,
+		@IExtensionService private readonly extensionService: IExtensionService
 	) {
 		super();
+		// Read in the cache on statup
+		this.cache = new Set<string>(JSON.parse(this.storageService.get(EditorOverrideService.overrideCacheStorageID, StorageScope.GLOBAL, JSON.stringify([]))));
+		this.storageService.remove(EditorOverrideService.overrideCacheStorageID, StorageScope.GLOBAL);
+
+		this._register(this.lifeCycleService.onWillShutdown(() => {
```

Comment:
While this works and ensures to run only once on shutdown, there is also `IStorageService#onWillSaveState` which is more specific to the storage service. This method is called on shutdown as well but also in other intervals to prevent loss of state in the case of a crash. E.g. this method may be called when the user moves focus out of the application to anther window.